File: java/log-levels/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
        int colonIndex = logLine.indexOf(":");
        return logLine.substring(colonIndex+1,logLine.length()).trim();
    }

    public static String logLevel(String logLine) {
        int colonIndex = logLine.indexOf(":");
        return logLine.substring(1,colonIndex-1).toLowerCase();
    }

    public static String reformat(String logLine) {
        return message(logLine) + " (" + logLevel(logLine) + ")";
    }
}

File: csharp/log-levels/LogLevels.cs
using System;

static class LogLine
{
    public static string Message(string logLine)
    {
        int targetIndex = logLine.IndexOf(':');
        string msg = logLine[(targetIndex + 1)..];

        return msg.Trim();
    }

    public static string LogLevel(string logLine)
    {
        string level = "";
        int targetIndex = logLine.IndexOf(':');
        string l = logLine[..targetIndex];
        char[] chars = l.ToCharArray();

        foreach (char c in chars)
        {
            if (Char.IsLetter(c))
            {
                level += c.ToString();
            }
        }

        return level.ToLower();
    }

    public static string Reformat(string logLine)
    {
        return $"{Message(logLine)} ({LogLevel(logLine)})";
    }
}
